packages/app/features/board/index.js

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import "./style.css"
+
+function Square({ value, onSquareClick }) {
+    return (
+        <button
+            className="square"
+            onClick={onSquareClick}
+        >
+            {value}
+        </button>
+    );
+}
+
+export default function Board() {
+
+    let[xTurn, setXTurn] = useState(true);
+    
+    let [sqrValues, setSqrValues] = useState(Array(9).fill(null));
+
+    function handleClick(i) {
+        if(sqrValues[i] || theWinnerIs(sqrValues)) { return; }
+
+        let nextValues = sqrValues.slice();
+        
+        if (xTurn) {
+            nextValues[i] = "X";
+        } else {
+            nextValues[i] = "O";
+        }
+
+        setSqrValues(nextValues);
+        setXTurn(!xTurn);
```

Comment:
Remember I told you the `set` functions are async? The logic you created could be a problem if the user taps twice very fast and ( hypothetically ) the page is very heavy and slow. The state would be unpredictable because you calculated the next state in this `handleClick` function with an old previousState in mind.

The way to do this right is to set the state by passing a FUNCTION, which receives the previousState and calculates the next state based on that. Doing it that way would make your code more error proof because it would work in an asynchronous manner.

TIP: The easiest way to refactor your code is to get rid of one of your states. The `xTurn` could be derived by how many squares are not null 😎 
Alternatively you could create your own type called `GameState` and have only one `useState`. By accessing your GameState you could easily change both:
``` tsx
enum Player {
    X,
    O,
}

interface GameStateType {
    board: SquareValue[];
    turn: Player;
    winner: Player | null; 
}

export default function Board() {
    const [gameState, setGameState] = useState<GameStateType>({
        board: [null, null, null, null, null, null, null, null, null, null],
        turn: Player.X,
        winner: null,
    });

    function handleClick(index: number) {
        setGameState((previousState) => {
            if(previousState.board[index] != null) {
                return previousState;
            }

            const newBoardState = [...previousState.board];
            newBoardState[index] = previousState.turn == Player.X ? "X" : "O";

            const newTurnState = previousState.turn == Player.X ? Player.O : Player.X;

            return {
                board: newBoardState,
                turn: newTurnState,
                winner: calculateWinner(newBoardState),
            }
        });
    }


    useEffect(() => {
        if(gameState.winner != null) {
            // Navigate to Winner screen
            // Pass the param `winner` to the navigation
        }
    }, [gameState])
    // The useEffect hook will be executed if the dependency array ( the last param , which is [gameState] )
    // changes. It always executes after the rendering ( in other words, after the "return" of your function )

   // ...
   /// return bla bla bla
}
```

Reply:
One last comment, your `handleClick` function will be recreated everytime there's a new render, right?

You can fix that by putting it inside a `useCallback` hook. This hook only recreates the function if something in the dependency array changes

Reply:
Done (I think)